src/Uno.Extensions.Core.Generators/PropertySelector/Rules.PropertySelector.cs

```diff
@@ -193,18 +173,18 @@ public static class PS0102
 			helpLinkUri: "https://aka.platform.uno/PS0102",
 			isEnabledByDefault: true);
 
-		public static Diagnostic GetDiagnostic(IMethodSymbol method, string param, SyntaxNode argNode)
+		private static Diagnostic GetDiagnostic(string method, string param, SyntaxNode argNode)
 			=> Diagnostic.Create(
 				Descriptor,
 				Location.Create(argNode.SyntaxTree, argNode.Span),
 				param,
-				method.ToString());
+				method);
 
-		public static GenerationException FailFileArg(IMethodSymbol method, IParameterSymbol? param, SyntaxNode argNode)
-			=> new(GetDiagnostic(method, "[CallerFilePath] " + (param?.Name ?? ""), argNode));
+		public static Diagnostic FailFileArg(IMethodSymbol method, IParameterSymbol? param, SyntaxNode argNode)
```

Comment:
```suggestion
		public static Diagnostic GetFileArgDiagnostic(IMethodSymbol method, IParameterSymbol? param, SyntaxNode argNode)
```

As you removed the `Fail` methods we can consider to renamed all `GetDiagnostic` to have something shorter, but we need to stay consistent :)